runner/database_neo4j.go

```diff
@@ -0,0 +1,57 @@
+package runner
+
+import (
+	"io"
+
+	jsonutil "github.com/multiprocessio/go-json"
+	"github.com/neo4j/neo4j-go-driver/v4/neo4j"
+)
+
+func (ec EvalContext) evalNeo4j(panel *PanelInfo, dbInfo DatabaseConnectorInfoDatabase, server *ServerInfo, w io.Writer) error {
+	_, conn, err := ec.getConnectionString(dbInfo)
+	if err != nil {
+		return err
+	}
+
+	password, err := ec.decrypt(&dbInfo.Password)
+	if err != nil {
+		return err
+	}
+
+	driver, err := neo4j.NewDriver(conn, neo4j.BasicAuth(dbInfo.Username, password, ""))
+	if err != nil {
+		return err
+	}
+	defer driver.Close()
+
+	sess := driver.NewSession(neo4j.SessionConfig{})
+
+	result, err := sess.Run(panel.Content, nil)
+	if err != nil {
+		return err
+	}
+
+	return withJSONArrayOutWriterFile(w, func(w *jsonutil.StreamEncoder) error {
+		records, err := result.Collect()
+		if err != nil {
+			return nil
+		}
+
+		for _, record := range records {
+			row := make(map[string]any)
+
+			for _, key := range record.Keys {
+				value, ok := record.Get(key)
+				if ok {
```

Comment:
Why is this sometimes not ok? If it's not ok we should probably still set `row[key] = nil`

Reply:
Because neo4j doesn't store null values, if you set something to null it will just delete it.

Reply:
Gotcha, well then as long as `value` comes back `nil` if it doesn't exist then let's do:

```
row[key], _ := record.Get(key)
```

And drop the ok check since DataStation always needs a value whether it's nil or not.